pkg/network/setup/configstate.go

```diff
@@ -49,7 +54,7 @@ func NewConfigState(configStateCache configStateCacheReaderWriter, ns NSExecutor
 //
 // The discovery step can be executed repeatedly with no limitation.
 // The configuration step is allowed to run only once. Any attempt to run it again will cause a critical error.
-func (c ConfigState) Run(nics []podNIC, preRunFunc func([]podNIC) ([]podNIC, error), discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
+func (c *ConfigState) Run(nics []podNIC, preRunFunc func([]podNIC) ([]podNIC, error), discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
```

Comment:
Does `ConfigState` still has to be a pointer?
Sorry if I ask this again but I lost track of how we address configState, the only thing that seems to be mutated is `cache` attribute which is a reference type.
Isnt it means that if cache is mutated it won't be discarded because it's a reference type?

Reply:
It doesn't have to be a pointer, now it will work if if it is not a pointer. But I prefer it to be a pointer to avoid situations like we had before I removed the podInterfaceName cache.
If it is pointer, changes to any future fields that will be added to ConfigState (no matter if the fields are pointer or not, overridden or not) will be reflected in the ConfigState stored in the NetConf.